src/stasis/core.clj

```diff
@@ -113,8 +112,11 @@
           (ensure-valid-paths (keys pages))
           (if-let [pageish (pages (:uri request))] ;; a pageish is either a page, or a function that creates a page
             (let [page (realize-page pageish request)]
-              (populate-known-dependent-pages (:uri request) page known-dependent-pages)
-              (serve-page page (:uri request)))
+              (if (and (nil? page) (:stasis/ignore-nil-pages? options))
+                not-found
```

Comment:
I suggest always taking the first branch when `(nil? page)` and then either returning `not-found` when the config option is set, otherwise giving a useful error message. Something like:

```clj
(if (nil? page)
  (if (:stasis/ignore-nil-pages? options)
    not-found
    (throw (ex-info "Page function unexpectedly returned nil" {:uri (:uri request)})))
  ,,,)
```

Thereby adressing both case 1) and 2) in the initial issue.